SPEC.md

```diff
@@ -924,6 +927,6 @@ JSON Schema for the Pod Manifest, conforming to [RFC4627](https://tools.ietf.org
     * **readOnly** (boolean, optional if **kind** is "host", defaults to "false" if unsupplied) whether or not the volume will be mounted read only.
 * **isolators** (list of objects, optional) list of isolators that will apply to this pod. Each object has two key value pairs: **name** is restricted to the [AC Name](#ac-name-type) formatting and **value** can be a freeform string)
 * **annotations** (list of objects, optional) arbitrary metadata the executor will make available to applications via the metadata service. Objects must contain two key-value pairs: **name** is restricted to the [AC Name](#ac-name-type) formatting and **value** is an arbitrary string). Annotation names must be unique within the list.
-* **ports** (list of objects, optional) list of ports that will be exposed on the host.
-    * **name** (string, required) name of the port in the image manifest that will be exposed on the host (restricted to the [AC Name](#ac-name-type) formatting).
-    * **hostPort** (integer, required) port number on the host that will be mapped to the container port.
+* **ports** (list of objects, optional) list of ports that SHOULD be exposed on the host.
+    * **name** (string, required, restricted to the [AC Name](#ac-name-type) formatting) name of the port to be exposed on the host. This field is a key referencing by name ports specified in the Image Manifest(s) of the app(s) within this Pod Manifest; consequently, port names MUST be unique among apps within a pod.
```

Comment:
I consider this a real shortcoming of kubernetes, frankly, and I wanted to fix it but it was hard, so we punted.  We may never get to fix it.

Specifically, you want containers to be arbitrarily composable but you have introduced an a priori unresolvable conflict.  If two containers both expose a port "http", your pod is invalid.  It's historically OK in kubernetes because Docker doesn't name ports, so it's the pod spec that names them, which is in the control of the pod deployer.  If you embed port names into images, we lose that control.

Options:
1a. always reference port names as appname.portname
1b. allow appname.portname but allow bare portname if there's no conflict
2a. demand that pod authors expose and name app ports in the pod spec
2b. allow pod authors to re-name app ports in the pod spec but allow pass-through if there's no conflict

others?